shell/browser/native_window_views_win.cc

```diff
@@ -28,6 +28,53 @@ namespace electron {
 
 namespace {
 
+void SetWindowBorderAndCaptionColor(HWND hwnd, COLORREF color) {
+  if (base::win::GetVersion() < base::win::Version::WIN11)
+    return;
+
+  HRESULT result =
+      DwmSetWindowAttribute(hwnd, DWMWA_CAPTION_COLOR, &color, sizeof(color));
+
+  if (FAILED(result))
+    LOG(WARNING) << "Failed to set caption color";
+
+  result =
+      DwmSetWindowAttribute(hwnd, DWMWA_BORDER_COLOR, &color, sizeof(color));
+
+  if (FAILED(result))
+    LOG(WARNING) << "Failed to set border color";
+}
+
+DWORD GetAccentColor() {
```

Comment:
```suggestion
std::optional<DWORD> GetAccentColor() {
```

If the `RegOpenKeyEx` call fails, currently this would return `0` which would then most likely result in a border color of black? So seems best to guard against that case and ensure the returned value from `GetAccentColor` is valid before applying it.

Reply:
Would also recommend to replace calls with the base utility for dealing with registry https://source.chromium.org/chromium/chromium/src/+/main:base/win/registry.h;l=24-33;